cpp/example/src/process_csv.cpp

```diff
@@ -0,0 +1,69 @@
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/per_device_resource.hpp>
+
+#include <cudf/aggregation.hpp>
+#include <cudf/groupby.hpp>
+#include <cudf/io/csv.hpp>
+#include <cudf/table/table.hpp>
+
+std::unique_ptr<cudf::table> read_csv(std::string const& file_path)
+{
+  auto source_info    = cudf::io::source_info(file_path);
+  auto builder        = cudf::io::csv_reader_options::builder(source_info);
+  auto options        = builder.build();
+  auto data_with_meta = cudf::io::read_csv(options);
+  return std::move(data_with_meta.tbl);
```

Comment:
```suggestion
  return data_with_meta.tbl;
```

Reply:
I think `move` is required here because the struct `data_with_meta` still owns the the `unique_ptr`. Similar to why we need a move for a `unique_ptr` in a vector.
example: https://godbolt.org/z/T9x5qWfYa